src/Hadolint/Rule/DL3058.hs

```diff
@@ -0,0 +1,27 @@
+module Hadolint.Rule.DL3058 (rule) where
+
+import qualified Data.Map as Map
+import Text.Email.Validate as Email
+import Hadolint.Rule
+import Language.Docker.Syntax
+import Data.Text.Encoding
+
+
+rule :: LabelSchema -> Rule args
+rule labelschema = mconcat $
+  fmap labelIsNotEmailRule (Map.keys (Map.filter (== Email) labelschema))
+{-# INLINEABLE rule #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+
+labelIsNotEmailRule :: LabelName -> Rule args
+labelIsNotEmailRule label = simpleRule code severity message check
+  where
+    code = "DL3058"
+    severity = DLWarningC
+    message = "Label `" <> label <> "` is not a valid email format - must be conform to RFC5322."
+    check (Label ls) = null $ getBadEmailformatLabels label ls
+    check _ = True
+
+getBadEmailformatLabels :: LabelName -> Pairs -> Pairs
+getBadEmailformatLabels lbl pairs = [(l, v) | (l, v) <- pairs, l == lbl, not (Email.isValid (encodeUtf8 v))]
```

Comment:
Just a suggestion: `isValid` only accepts fully compliant strings. If you want to allow something like `user . name @ host . com` as well (with spaces), the `Text.Email.Validate` module has a `canonicalizeEmail` function, which can be used for that. I don't know if its still a common practice to obfuscate emails like that to avoid spam, but `canonicalizeEmail` would allow that too.

Reply:
I dont have a use case for obfuscated emails and I personally dont like to make code change for such a rare use case that might or might not come.
how about we leave it be for now and if there is an ask, we can add it ? 
Or if you would like that added, I can definitely add that change too.

Reply:
Fair enough, it was just a suggestion.